File: Python/Math-Array-String/11.py
# 1979. Find Greatest Common Divisor of Array


from typing import List

class Solution:
    def findGCD(self, nums: List[int]) -> int:
        mn = min(nums)
        mx = max(nums)
        for i in range(1, mn+1):
            if mn % i == 0 and mx % i == 0:
                gcd = i
        return gcd
    

sol = Solution()
nums = [3,3]
output = sol.findGCD(nums)
print(output)


# import math
# class Solution:
#     def findGCD(self, nums: List[int]) -> int:
#         mn = min(nums)
#         mx = max(nums)
#         gcd = math.gcd(mn, mx)
#         return gcd

File: JavaScript/Math-Array-String/11.js
// 1979. Find Greatest Common Divisor of Array

/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    let ma = Math.max(...nums)
    let mi = Math.min(...nums)

    let gcd = 1
    for(let i=1; i<=mi; i++) {
        if(ma % i === 0 && mi % i === 0) {
            gcd = i;
        }
    }
    return gcd;
};

console.log(findGCD([3, 3]));
